src/Component/Details.js

```diff
@@ -1,11 +1,52 @@
 import React from 'react';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
 import '../Style/details.css';
 
-const Details = () => (
-  <div>
-    Details
-    <h1>hhbhufbghgjph</h1>
-  </div>
-);
+const Details = () => {
+  const { fName } = useParams();
+
+  const { TheCountryData } = useSelector((state) => state.country);
+  const selectedCountry = TheCountryData.find((item) => item.name === fName);
+
+  return (
+    <>
+      <div className="details-container">
+        <div className="image-container">
+          <h1 className="header-text">{selectedCountry?.name || 'none'}</h1>
+          <img src={selectedCountry.flag} alt={selectedCountry.name} className="flagofthe-country" />
```

Comment:
- [ ] You have written your code well but there are three tests that are failing in your tests. and all three breaks while testing this line. so kindly consider passing all the tests to improve the code maintainability.
<img width="514" alt="Screenshot 2023-05-30 at 10 25 50 PM" src="https://github.com/KAYEMBAJONATHAN/React-Capstone/assets/96683843/4bc2f1c3-9a08-4bb2-a85f-6ee814aa8898">
<img width="982" alt="Screenshot 2023-05-30 at 10 26 01 PM" src="https://github.com/KAYEMBAJONATHAN/React-Capstone/assets/96683843/2e46f3b3-06c0-4431-ae24-f624db5613ee">